plugins/eventMacro/eventMacro/Condition/IsInMapAndCoordinate.pm

```diff
@@ -17,62 +19,150 @@ my $map_qr = qr/\w+\d*_?(?:\d|\w)*/;
 sub _parse_syntax {
 	my ( $self, $condition_code ) = @_;
 	
-	$self->{validators_index} = {};
-	$self->{member_index_to_validator_indexes} = [];
 	$self->{is_on_stand_by} = 0;
+	
 	$self->{fulfilled_coordinate} = undef;
 	$self->{fulfilled_member_index} = undef;
 	
-	my $member_counter = 0;
-	my $array_counter = 0;
-	my @members = split(/\s*,\s*/, $condition_code);
-	my @coordinates_array;
-	foreach my $member (@members) {
+	$self->{var_to_member_index_coord_x} = {};
+	$self->{var_to_member_index_coord_y} = {};
+	$self->{var_to_member_index_map} = {};
+	
+	$self->{x_validators} = {};
+	$self->{y_validators} = {};
+	$self->{map_validators} = {};
+	
+	my $var_exists_hash = {};
+	
+	my $member_index = 0;
+	foreach my $member (split(/\s*,\s*/, $condition_code)) {
 		my ($map, $coord_x, $coord_y);
 		if ($member =~ /^(?:($map_qr)|(?:(\S+)\s+(\S+))|($map_qr)\s+(?:(\S+)\s+(\S+)))$/) {
```

Comment:
map doesnot support variables because of this regex i think.
```
[eventMacro] Ignoring automacro 'IsInMapAndCoordinate'
[eventMacro] Error in condition 'IsInMapAndCoordinate'
[eventMacro] Error type: Wrong condition syntax ('$map 97 62, 96 62, $map 96 61')
[eventMacro] Error code: 'List member '$map 97 62' must have a map and/or a coordinate'.
```

Reply:
but x and y are supporting variables normally